src/provisioner/provisioner.py

```diff
@@ -101,7 +107,18 @@ def config_apply(solution_config_url: str, cortx_conf_url: str = None,
         if Conf.get(CortxProvisioner._solution_index, 'cluster') is not None:
             CortxProvisioner.apply_cluster_config(cortx_conf, CortxProvisioner.cortx_release)
 
-        if Conf.get(CortxProvisioner._solution_index, 'cortx') is not None:
+        Machine_id = CortxProvisioner._get_machine_id()
+        if not CortxProvisioner._wait_for_lock_to_be_released(cortx_conf, CortxProvisioner._timeout):
```

Comment:
I suggest wait_for_lock should be an infinite loop which  breaks on the condition
 if lock_time + timeout > current time break the loop and force unlock

This is to satisfy the condition what if the first holder could not complete the job in time and second node  unlocks forcefully, the remaining nodes do not follow the force unlock part

Reply:
done